### ISSUE TITLE
Removal of the ossec-init.conf file from the Wazuh installations - Analysis

### ISSUE THREAD

**Issue by DProvinciani** [wazuh/wazuh#7055]
### Description

As part of the epic https://github.com/wazuh/wazuh/issues/7054, the `ossec-init.conf` file should be removed from the Wazuh installations. Considering the requirements listed in the epic description, a complete analysis should be made focusing on all the things that should be modified/updated.

As a result of this issue, we should have a clear understanding of all the changes that should be implemented as part of the epic and a list of issues with an effort estimated.

**Comment by DProvinciani** [comment]
As a result of the analysis that we made, we got the next list of tasks to be implemented.

- **TASK 0**: Create the environment variable `WAZUH_HOME`
  - It should be readable by all the users.
  - It should be persistent after reboots.
  - It should be tested in all the operating systems supported by Wazuh.

- **TASK 1**: Instalacion related changes
  - Change the name of the file to `wazuh-init.conf`.
  - install.sh: This script should create the file in `$WAZUH_HOME/etc`, and the symbolic link shouldn't be longer created.
  - Init files for service management:
    - src/init/darwin-init.sh: Init para MACOS?
    - src/init/ossec-hids-aix.init
    - src/init/ossec-hids-debian.init
    - src/init/ossec-hids-gentoo.init
    - src/init/ossec-hids-hpux.init
    - src/init/ossec-hids-rh.init
    - src/init/ossec-hids-solaris.init
    - src/init/ossec-hids-suse.init
    - src/init/ossec-hids.init
    - src/init/pkg_installer.sh
    - src/init/register_configure_agent.sh
    - src/init/shared.sh (verify the right behavior of the next scripts after implementing the changes... `add_localfiles.sh`, `gen_ossec.sh`, `install.sh`, `inst-functions.sh`)
    - src/init/wazuh-client.sh
    - src/init/wazuh-local.sh
    - src/init/wazuh-server.sh
    - src/systemd/wazuh-agent.service
    - src/systemd/wazuh-manager.service
  - Archivo BUGS para que refiera a $WAZUH_HOME/etc/ossec-init.conf

- **TASK 2**: Upgrade related changes
  - upgrade.sh: Make use of `$WAZUH_HOME`. If it doesn't exist, verify if the `ossec-init.conf` file exists (in this case, the upgrade should be done, then create the `$WAZUH_HOME` variable and finally, remove the file). If the variable nor the file exists, the upgrade shouldn't be done.
  - update.sh
  - pkg_installer.sh
  - src/init/inst-functions.sh: This file generates the new `ossec-init.conf`. It should be modified to instead generate `wazuh-init.conf`.

- **TASK 3**: Update the analysisd related scripts
  - src/analysisd/compiled_rules/register_rule.sh
  - ruleset/testing/runtests.py

- **TASK 4**: Update the kaspersky related scripts:
  - active-response/kaspersky.py
  - active-response/kaspersky.sh

- **TASK 5**: Update the next utils to read the `wazuh-init.conf` file taking into account the variable `$WAZUH_HOME`:
  - contrib/ossec-eps.sh
  - contrib/util.sh
  - The next files will be analyzed and evaluated whether to keep them or deprecate them:
    - contrib/rename_agent.sh
    - contrib/renumber_agent.sh

- **TASK 6**: AWS and gcloud related scripts
  - wodles/aws/aws_s3.py
  - wodles/aws/tests/test_aws.py
  - wodles/gcloud/tools.py
  - wodles/gcloud/tests/common.py
  - Clean the `directory` variable from the `wazuh-init.conf` test files.

- **TASK 7**: Update the docker listener script
  - wodles/docker-listener/DockerListener.py

- **TASK 8**: Analizar si deben modificarse:
  - Update the script `gen_ossec.sh`.
  - Evaluate whether the next file should be updated or deprecated:
    - contrib/debian-packages/ossec-hids/debian/postinst
    - contrib/debian-packages/ossec-hids/debian/postrm
    - contrib/debian-packages/ossec-hids-agent/debian/postinst
    - contrib/debian-packages/ossec-hids-agent/debian/postrm

- **TASK 9**: Tools:
  - Run files and `run.sh`. Only used to get the directory.
  - `aws.py`: Only gets the directory.
  - `check_files.py`: The file attributes should be removed.
  - `template_manager_v2.0.1.json` and `template_manager_v2.0.1.json` contains the file attributes. They should be removed.
  - `analysisd_measure.sh`: Only gets the directory.
  - `remoted_measure`: Only gets the directory.
  - `check_files.sh`: It just check file permissions. It should be removed.
  - `clean-agents`: Only gets the directory and version.
  - `restart_when_pushed.sh`: Only gets the directory and version.
  - `wazuh shell`: Only gets the directory for the uninstallation. It should be verified whether is still functional and if not it should be removed.
  - `upgrade.sh`: Only gets the directory.
  - `upgrade_agent`: Only gets the version.
  - `remove_alerts_logs.sh`: Only gets the directory and version.
  - `wazuh_authd_deb`: Only gets the directory.
  - `wazuh_auth_systemctl`: Only gets the directory and name.
  - `boot_authd.sh`: Only gets the directory.
  - `manager-deb.sh`: Only gets the version.
  - `manager-repm.sh`: Only gets the version.

- **TASK 10**: Docu:
  - Modify the example in https://documentation.wazuh.com/4.0/user-manual/capabilities/file-integrity/fim-configuration.html#configuring-syscheck-basic-usage to avoid using `ossec-init.conf`. It should only read the root environment variables. Verify the right behavior by restarting with both, `systemd` and `wazuh-control`.
  - https://documentation.wazuh.com/4.0/installation-guide/more-installation-alternatives/wazuh-from-sources/wazuh-agent/index.html
  - https://documentation.wazuh.com/4.0/migrating-from-ossec/ossec-agent.html?highlight=ossec%20init
  - https://documentation.wazuh.com/4.0/migrating-from-ossec/ossec-server.html?highlight=ossec%20init
  - Update all the uninstallation instructions.

- **TASK 11**: Exploratory testing.

- **TASK 12**: Tracking issue for creating all the PRs and documentation of the epic.


### DEPENDENCIES

- **Framework**:
  - framework/docs/source/wazuh_framework_getting_started.rst
  - framework/scripts/wazuh-logtest.py

- **Packages**:
  - The installations should create the `$WAZUH_HOME` variable.
  - The uninstallations should remove the `$WAZUH_HOME` variable.
  - The upgrades should create the `$WAZUH_HOME` is it does not exist.

- **QA**:
  - `/tools/__init__.py`: It only takes the type from the file.
  - `test_wpk_agent` and `test_wpk_manager` make use of the file to know the agent version installed.
  - `sca_directories_test_suite.yaml` contains some test cases that make use of the file.

**Comment by DProvinciani** [comment]
After starting with the implementation of this approach, we found some problems that could result in a bas user experience as well as in some extra complexity of the installation process. Having said that, we decided to analyze some other possibilities in order to avoid these problems.

In the issue https://github.com/wazuh/wazuh/issues/7129 we provide details about the problems that we found and also the description of the new design that will be implemented.
